packages/type-utils/tests/isTypeReadonly.test.ts

```diff
@@ -0,0 +1,259 @@
+import * as ts from 'typescript';
+import { TSESTree } from '@typescript-eslint/experimental-utils';
+import { parseForESLint } from '@typescript-eslint/parser';
+import {
+  isTypeReadonly,
+  type ReadonlynessOptions,
+} from '../src/isTypeReadonly';
+import path from 'path';
+
+describe('isTypeReadonly', () => {
+  const rootDir = path.join(__dirname, 'fixtures');
+
+  describe('TSTypeAliasDeclaration ', () => {
+    function getType(code: string): {
+      type: ts.Type;
+      checker: ts.TypeChecker;
+    } {
+      const { ast, services } = parseForESLint(code, {
```

Comment:
Hmm, this is essentially the same as the code at the beginning of `isUnsafeAssignment`. 

https://github.com/typescript-eslint/typescript-eslint/blob/a9eb0b9eb2db291ea36065ec34f84bf5c5504b43/packages/type-utils/tests/isUnsafeAssignment.test.ts#L16

Since we're starting to establish a precedent of unit testing these type utilities directly, I think we'll want to make a standard way to generate the AST & node maps for them. What do you think about making a standard util like this?:

```ts
const { body, checker, esTreeNodeToTSNodeMap } = parseForUtilTest(code);
```